recipes/openvdb/all/conanfile.py

```diff
@@ -27,91 +30,202 @@ class OpenVDBConan(ConanFile):
 
     @property
     def _compilers_min_version(self):
-        return {
-            "msvc": "191",
-            "Visual Studio": "15",  # Should we check toolset?
-            "gcc": "6.3.1",
-            "clang": "3.8",
-            "apple-clang": "3.8",
-            "intel": "17",
-        }
+        if Version(self.version) == "8.0.1":
+            return {
+                "msvc": "191",
+                "Visual Studio": "15",  # Should we check toolset?
+                "gcc": "6.3.1",
+                "clang": "3.8",
+                "apple-clang": "3.8",
+                "intel": "17",
+            }
+        elif Version(self.version) == "10.0.1":
+            return {
+                "msvc": "1928",
+                "Visual Studio": "16.8.1",  # Should we check toolset?
+                "gcc": "9.3.1",
+                "clang": "5",
+                "apple-clang": "5",
+                "intel": "19",
+            }
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if not self._has_sse_avx:
+            del self.options.simd
+        
+        if Version(self.version) < "9.0.0":
+            del self.options.nanovdb
+            del self.options.nanovdb_with_openvdb
+            del self.options.nanovdb_with_tbb
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def _cmake_new_enough(self, required_version):
+        try:
+            import re
+            from io import StringIO
+            output = StringIO()
+            self.run("cmake --version", output=output)
+            m = re.search(r'cmake version (\d+\.\d+\.\d+)', output.getvalue())
+            return Version(m.group(1)) >= required_version
+        except:
+            return False
+
+    def build_requirements(self):
+        if Version(self.version) >= "10.0.0" and not self._cmake_new_enough("3.18"):
+            self.tool_requires("cmake/[>=3.18.0]")
+        elif Version(self.version) >= "9.0.0" and not self._cmake_new_enough("3.15"):
+            self.tool_requires("cmake/[>=3.15.0]")
+        elif not self._cmake_new_enough("3.12"):
+            self.tool_requires("cmake/[>=3.12.0]")
 
     def requirements(self):
-        self.requires("boost/1.79.0")
-        self.requires("onetbb/2020.3")
-        self.requires("openexr/2.5.7")  # required for IlmBase::Half
+        if self._needs_boost:
+            self.requires("boost/1.80.0")
+
+        if Version(self.version) < "9.0.0":
+            self.requires("onetbb/2020.3")
```

Comment:
This PR is failing for 2.0 because ``onetbb/2020.3`` has not been ported to Conan 2.0.

The new cmake-based ``onetbb/2021`` have been ported, is it possible to upgrade this recipe so it requires a more modern ``onetbb``?

Reply:
I try to follow the recommended versions from their [repository](https://github.com/AcademySoftwareFoundation/openvdb/blob/v10.0.1/doc/dependencies.txt) upgrading when possible. In this case using a newer version is not supported since until version 9 `OpenVDB` was relying on `tbb::atomic` and other implementations that were replaced by the `std` equivalents in `onetbb/2021`. This is the main reason I'm upgrading to v10 because right now I can't use `OpenVDB` and `Embree` with the same `onetbb` version